pkg/watcher/internal/metricsprovider/prometheus.go

```
@@ -61,10 +73,73 @@ func NewPromClient(opts watcher.MetricsProviderOpts) (watcher.MetricsProviderCli
 		promAddress = opts.Address
 	}
 
+	// Ignore TLS verify errors if InsecureSkipVerify is set
+	roundTripper := api.DefaultRoundTripper
+
+	// Check if EnableOpenShiftAuth is set.
+	_, enableOpenShiftAuth := os.LookupEnv(EnableOpenShiftAuth)
+	if enableOpenShiftAuth {
+		// Create the config for kubernetes client
+		clusterConfig, err := rest.InClusterConfig()
+		if err != nil {
+			// Get the kubeconfig path
+			kubeConfigPath, ok := os.LookupEnv(kubeConfig)
+			if !ok {
+				kubeConfigPath = defaultKubeConfig
+			}
+			clusterConfig, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
```

This `err` is swallowed